setup.py

```diff
@@ -63,6 +42,101 @@ class CustomBuild(build):
     ]
 
 
+def find_glpk_header():
+    # If a path is provded by the environment, expect the GLPK header there.
+    given_path = os.environ.get(ENV_GLPK_HEADER_PATH, None)
+    if given_path:
+        if not os.path.isdir(given_path):
```

Comment:
This is a bit different from the previous logic and is what is breaking the windows builds here. I think it's okay to check the other locations if the env variable is not a valid directory, so I would not throw an error here yet.

Reply:
What does it point to? The header file itself?

Reply:
If under Windows the variable points to the header file itself than my update should make that work, otherwise I'm not sure why this would happen as I can't see any logs. If need be I can remove the check but I find it sensible that a given path should not be silently ignored if it has a typo or so.

Reply:
Your handling of the path is all good. On windows it also points to a directory but in the CI this directory defaults to `glpk-5.0/src` which is not a valid directory. Admittedly, this is an issue with the CI but it does expose that the new version behaves different. In the old script it would just jump to looking for `glpk.h` in the current directory afterwards but in the new one it errors and stops. I think the old behavior made a bit more sense. If the env variable does not make sense it should still try to find it in the current directory. It could show a warning but it's too early to throw an error and give up.

Reply:
Alright, I will print a warning then.